### DISCUSSION TITLE
How do we Cache only success(200) response from micro services via API Gateway?

### DISCUSSION THREAD

**Discussion by chitharansivasamy** [ThreeMammals/Ocelot#1890] [Q&A]
## Expected Behavior / New Feature

Ocelot API Gateway is supposed to caching only when gets success responses (200) from micro-services

## Actual Behavior / Motivation for New Feature
The problem is we had used the below Ocelot JSON to cache the response and it's holding the data for one full day (86400 secs). here is suppose to get some errors during the services call like(500,502,499...). what happened is the cached error response was not even updated till one full day even the services issue got resolved.

"UpstreamPathTemplate": "/Products/GetProductMetaData",
"UpstreamhttpsMethod": [ "GET" ],
"FileCacheOptions": { "TtlSeconds": 86400 },
So we are looking for the gateway to cache only success and not the failure cases. It should be hit services till when its bring success(200).
Is there any option in OCELOT to restrict caching based on the HTTP Status code like 500, 502, 503, 499..etc?

**Comment by andyjhall** [comment]
Since the response is being cached with cachemanager id guess (and it is a guess) that youd need to check the response inside here and if its not successful then not add that response to the cache:

https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot.Cache.CacheManager/OcelotCacheManagerCache.cs

`public void Add(string key, T value, TimeSpan ttl, string region)`
`        {`
`            _cacheManager.Add(new CacheItem<T>(key, region, value, ExpirationMode.Absolute, ttl));`
`        }`

**Comment by elie-chaaban** [comment]
how did you fix it

**Reply by raman-m** [reply]
Don't worry!
We will find a solution...
First of all, we need to decrease caching time. 😄

**Comment by raman-m** [comment]
@chitharansivasamy 
Hi Chitharan!
The 1 day caching time is abnormally long!
Try to decrease this time down to 1 or 5 minutes, and it seems your problem will be solved. 😉
